### PR TITLE
New: Added Container element [ED-2609]

### PR DESCRIPTION
**Pull request by StyleShit**

### CODE REVIEW

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]

**Comment by StyleShit** [comment]
@matipojo Let's continue the discussion here:
https://github.com/elementor/elementor/pull/16655

File: assets/dev/js/editor/editor-base.js [hunk 1]
```diff
@@ -1375,8 +1375,7 @@ export default class EditorBase extends Marionette.Application {
 						controlArgs.default = controlArgs[ device + '_default' ];
 					}
 				} else if ( deleteControlDefault ) {
-					// In the Editor, controls without default values should have an empty string as the default value.
-					controlArgs.default = '';
+					delete controlArgs.default;
```
[the remark this answers is not in the document]
Comment (matipojo) [reply assets/dev/js/editor/editor-base.js:1391]: Please remove from this PR

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js [hunk 2]
```diff
@@ -62,7 +62,9 @@ PanelElementsLayoutView = Marionette.LayoutView.extend( {
 	},
 
 	initElementsCollection: function() {
-		const elementsCollection = new PanelElementsElementsCollection();
+		var elementsCollection = new PanelElementsElementsCollection();
```
Comment (matipojo) [inline assets/dev/js/editor/regions/panel/pages/elements/elements.js:65]:
```suggestion
		const elementsCollection = new PanelElementsElementsCollection();
```

File: assets/dev/js/editor/utils/jquery-html5-dnd.js [hunk 3]
```diff
@@ -191,9 +191,15 @@
...
-			elementsCache.$placeholder[ insertMethod ]( currentElement );
+			// Fix placeholder placement for Container with `flex-direction: row`.
+			if ( ! $( currentElement ).hasClass( 'elementor-first-add' ) && $( currentElement ).parents( '.e-container--placeholder-row' ).length ) {
+				const insertMethod = [ 'bottom', 'right' ].includes( currentSide ) ? 'after' : 'before';
+				$( currentElement )[ insertMethod ]( elementsCache.$placeholder );
+				elementsCache.$placeholder.css( 'order', $( currentElement ).css( 'order' ) );
+			} else {
+				const insertMethod = 'top' === currentSide ? 'prependTo' : 'appendTo';
+				elementsCache.$placeholder[ insertMethod ]( currentElement );
+			}
```
Comment (matipojo) [inline assets/dev/js/editor/utils/jquery-html5-dnd.js:202]: Please make it more readable

File: assets/dev/scss/editor/preview/preview.scss [hunk 4]
```diff
@@ -34,6 +36,24 @@ html.elementor-html {
...
 .elementor-edit-area-active {
 
+	--primary-color: #{ $editor-info };
+	--secondary-color: #{ darken($editor-info, 20%) };
+	--outline-color: var( --primary-color );
+
+	.e-container {
+		// TODO: Use variable when available - waiting for final color from the Design.
+		--primary-color: #0098C7;
+		--secondary-color: #{ darken(#0098C7, 10%) };
```
Comment (matipojo) [inline assets/dev/scss/editor/preview/preview.scss:46]: Do we have the final colors?

File: assets/dev/scss/editor/preview/preview.scss [hunk 5]
```diff
@@ -16,6 +16,8 @@
 //Eyedropper Experiment
```
Comment (matipojo) [inline assets/dev/scss/editor/preview/preview.scss:16]: Please test it on RTL & Dark mode

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/experiments/manager.php [hunk 6]
```diff
@@ -316,6 +317,19 @@ private function add_default_features() {
...
 		] );
+
+		$this->add_feature( [
+			'name' => 'container',
+			'title' => __( 'Container', 'elementor' ),
+			'description' => __(
+				'Create advanced layouts and responsive designs using the new Container element.
+				When this experiment is active, Containers will be the default building block.
+				Existing Sections, Inner Sections and Columns will not be affected.',
+				'elementor'
```
Comment (rodolphebertozzo) [inline core/experiments/manager.php:328]:
```suggestion
			'title' => esc_html__( 'Container', 'elementor' ),
			'description' => esc_html__(
				'Create advanced layouts and responsive designs using the new Container element.
				When this experiment is active, Containers will be the default building block.
				Existing Sections, Inner Sections and Columns will not be affected.',
				'elementor'
```

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]

File: assets/dev/js/editor/editor-base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/editor-base.js:1391]: Done

File: assets/dev/js/editor/regions/panel/pages/elements/elements.js (hunk 2 above)
Comment (StyleShit) [reply assets/dev/js/editor/regions/panel/pages/elements/elements.js:65]: Done

File: assets/dev/js/editor/utils/jquery-html5-dnd.js (hunk 3 above)
Comment (StyleShit) [reply assets/dev/js/editor/utils/jquery-html5-dnd.js:202]: Done

File: assets/dev/js/editor/components/template-library/views/parts/import.js [hunk 7]
```diff
@@ -59,7 +59,7 @@ TemplateLibraryImportView = Marionette.ItemView.extend( {
 			},
 		};
 
-		elementorCommon.ajax.addRequestConstant( 'uploadTypeCaller', 'elementor-media-upload' );
+		elementorCommon.ajax.addRequestConstant( 'uploadTypeCaller', 'elementor-wp-media-upload' );
```
[the remark this answers is not in the document]
Comment (StyleShit) [reply assets/dev/js/editor/components/template-library/views/parts/import.js:62]: Looks like in 3.5.0 it's the top one:
https://github.com/elementor/elementor/blob/release/3.5.0/assets/dev/js/editor/components/template-library/views/parts/import.js#L61-L63

File: assets/dev/scss/editor/preview/preview.scss (hunk 5 above)
Comment (StyleShit) [reply assets/dev/scss/editor/preview/preview.scss:16]: Done

File: assets/dev/scss/editor/preview/preview.scss (hunk 4 above)
Comment (StyleShit) [reply assets/dev/scss/editor/preview/preview.scss:46]: Nope.
Design said it's final for now.

**Comment by StyleShit** [comment]
@matipojo #16873

File: includes/autoloader.php [hunk 8]
```diff
@@ -195,6 +195,10 @@ private static function normalize_class_name( $string, $delimiter = ' ' ) {
 
 	private static function init_classes_aliases() {
 		self::$classes_aliases = [
+			'Editor' => [
+				'replacement' => 'Core\Editor\Editor',
+				'version' => '2.6.0',
+			],
```
Comment (matipojo) [inline includes/autoloader.php:201]: Remove (I think it's from a wrong merge)

File: assets/dev/scss/editor/preview/preview.scss (hunk 4 above)
Comment (matipojo) [reply assets/dev/scss/editor/preview/preview.scss:46]: Please set it on the SCSS variables file

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/autoloader.php (hunk 8 above)
Comment (StyleShit) [reply includes/autoloader.php:201]: Done

File: assets/dev/scss/editor/preview/preview.scss (hunk 4 above)
Comment (StyleShit) [reply assets/dev/scss/editor/preview/preview.scss:46]: Done

File: includes/elements/column.php [hunk 9]
```diff
@@ -1023,6 +1023,10 @@ protected function _get_default_child_type( array $element_data ) {
 			return Plugin::$instance->elements_manager->get_element_types( 'section' );
 		}
 
+		if ( 'container' === $element_data['elType'] ) {
+			return Plugin::$instance->elements_manager->get_element_types( 'container' );
+    }
```
Comment (matipojo) [inline includes/elements/column.php:1028]:
```suggestion
		if ( 'container' === $element_data['elType'] ) {
			return Plugin::$instance->elements_manager->get_element_types( 'container' );
		}
```

**Review by matipojo — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**